inductiva/resources/machines.py

```diff
@@ -24,188 +24,161 @@ def __init__(
-            return
-
-        try:
-            logging.info("Terminating machine group. "
-                         "This may take a few minutes.")
-            start_time = time.time()
-
-            request_body = \
-                inductiva.client.models.InstanceGroup(
-                    id=str(self.id),
-                    name=self.name,
-                    machine_type=self.machine_type,
-                    num_instances=self.num_machines,
-                    spot=self.spot,
-                    disk_size_gb=self.disk_size_gb,
-                    zone=self.zone,
-                )
-
-            self._api.delete_instance_group(body=request_body)
-            termination_time_mins = (time.time() - start_time) / 60
-            logging.info(
-                "Machine group of %s machines successfully "
-                "terminated in %.2f mins.\n", self.num_machines,
-                termination_time_mins)
-
-        except inductiva.client.ApiException as api_exception:
-            raise api_exception
-
-    def _get_estimated_cost(self) -> float:
-        if self._estimated_cost is not None:
-            return self._estimated_cost
-        instance_price = self._api.get_instance_price({
-            "machine_type": self.machine_type,
-            "zone": self.zone,
-        })
-        if self.spot:
-            estimated_cost = instance_price.body[
-                "preemptible_price"] * self.num_machines
-        else:
-            estimated_cost = instance_price.body[
-                "on_demand_price"] * self.num_machines
-
-        self._estimated_cost = float(round(estimated_cost, 3))
-        return self._estimated_cost
-
-    def estimate_cloud_cost(self) -> float:
-        """Returns the estimated cost per hour of a machine group.
-
-        Note that this is an estimate of the cost incurred in cloud resources,
-        and is not binding of the actual price of the machine group.
-        """
+        """Terminates all machines of the machine group."""
+        return super().terminate(num_instances=self.num_machines,
+                                 is_elastic=self.is_elastic)
+
+    def _log_machine_group_info(self):
+        super()._log_machine_group_info()
+        logging.info("> Number of machines: %s", self.num_machines)
+
+    def estimate_cloud_cost(self):
+        """Estimates a cost per hour of the machine group in US dollars.
+
+        This is only an estimate of having a machine group with the
+        specified configurations up in the cloud. The actual cost may vary.
+
+        Returns:
+            The estimated cost per hour of the machine group in US
+              dollars($/h)."""
         #TODO: Contemplate disk size in the price.
-        estimated_cost = self._get_estimated_cost()
-        logging.info("Estimated cloud cost per hour: %s $/h", estimated_cost)
+        estimated_cost = round(
+            float(super()._get_estimated_cost() * self.num_machines), 3)
```

Comment:
Why is this float required?

Reply:
Should the super() do the float and the round? In principle, there won't be a problem by multiplying here by the number of machines.

Reply:
super returns float, but the num_machines is not a float it has the schema type, so it threw an error when I only converted it to float in super.

Reply:
Then just float the number.